### PR TITLE
Upgrade to rails 5

### PR DESCRIPTION
**Pull request by jpascual1994**
This PR includes:
- Upgrade from rails 4.1 to 5.1

### CODE REVIEW

File: config/environments/test.rb [hunk 1]
```diff
@@ -25,12 +27,13 @@
...
   config.action_controller.allow_forgery_protection = false
+  config.action_mailer.perform_caching = false
 
   # Tell Action Mailer not to deliver emails to the real world.
   # The :test delivery method accumulates sent emails in the
   # ActionMailer::Base.deliveries array.
-  config.action_mailer.delivery_method = :letter_opener
-  config.action_mailer.perform_deliveries = true
+  config.action_mailer.delivery_method = :test
+  config.action_mailer.default_url_options = { :host => "localhost:3000" }
```
Comment (gabrielrp) [inline config/environments/test.rb]: Change to new rails syntax

Comment (gabrielrp) [reply config/environments/test.rb]: Use single quotes

File: config/secrets.yml [hunk 2]
```diff
@@ -5,18 +5,28 @@
 
 # Make sure the secret is at least 30 characters and all random,
 # no regular words or you'll be exposed to dictionary attacks.
-# You can use `rake secret` to generate a secure secret key.
+# You can use `rails secret` to generate a secure secret key.
```
Comment (gabrielrp) [inline config/secrets.yml]: You should ignore this file

File: spec/controllers/registrations_controller_spec.rb [hunk 3]
```diff
@@ -50,7 +50,10 @@
     context "with invalid params" do
       context "empty fields" do
         before(:each) do
-          post :create, user: { name: "", email: "", password: "", password_confirmation: "", gender: "male" }
+          post :create,
+            params: {
+              user: { name: "", email: "", password: "", password_confirmation: "", gender: "male" }
```
Comment (gabrielrp) [inline spec/controllers/registrations_controller_spec.rb]: You can define a function that has the data as entry and returns the params hash. Then use this function everywhere with the data you need.

**Review by gabrielrp — CHANGES_REQUESTED** [review]

**Review by gonzaM — APPROVED** [review]
I think the migration to rails 5 is ok.

File: spec/controllers/registrations_controller_spec.rb [hunk 4]
```diff
@@ -50,7 +50,10 @@
...
+            }
         end
 
         it "shouldn't create the user" do
```
Comment (gonzaM) [inline spec/controllers/registrations_controller_spec.rb]: Don't use should. Use for example: `'creates the user'`.
http://www.betterspecs.org/#should

File: spec/controllers/registrations_controller_spec.rb [hunk 5]
```diff
@@ -64,7 +67,10 @@
 
       context "with password too short" do
         before(:each) do
-          post :create, user: { name: "name", email: "mail@e", password: "1", password_confirmation: "1", gender: "male" }
+          post :create,
+            params: {
+              user: { name: "name", email: "mail@e", password: "1", password_confirmation: "1", gender: "male" }
```
Comment (gonzaM) [inline spec/controllers/registrations_controller_spec.rb]: Use single quotes (here and in the rest of the file)

File: spec/controllers/registrations_controller_spec.rb [hunk 6]
```diff
@@ -78,7 +84,10 @@
 
       context "with invalid password confirmation" do
         before(:each) do
-          post :create, user: { name: "name", email: "mail@e", password: "123456", password_confirmation: "qwerty", gender: "male" }
+          post :create,
+            params: {
+              user: { name: "name", email: "mail@e", password: "123456", password_confirmation: "qwerty", gender: "male" }
```
Comment (gonzaM) [inline spec/controllers/registrations_controller_spec.rb]: Some lines are too long.

**Review by gabrielrp — APPROVED** [review]

**Pull request merged**